src/MemBuf.h

```diff
@@ -25,13 +25,12 @@ class MemBuf : public Packable
     CBDATA_CLASS(MemBuf);
 
 public:
-    MemBuf():
-        buf(NULL),
-        size(0),
-        max_capacity(0),
-        capacity(0),
-        stolen(0)
-    {}
+    static const mb_size_t maxSize = (2*1000*1024*1024); ///< 2000 MB
+    static const mb_size_t defaultSize = (2*1024); ///< 2 KB
+
+    MemBuf() = default;
+    MemBuf(const MemBuf &) = delete;
+    MemBuf& operator= (const MemBuf &) = delete;
```

Comment:
That is two lines and moving the buffer is not a problem. Only copy-construct of a second MemBuf that uses the same memory allocation.

Reply:
> moving the buffer is not a problem. Only copy-construct of a second MemBuf that uses the same memory allocation.

The default-generated MemBuf move methods will crash Squid. Fortunately, they will not be generated in the current code (due to obscure set of rules that no human can remember). PR after PR, I am fixing and explaining the same move optimizations bugs. I understand that this is a difficult subject, but I really cannot understand why it is so hard to accept a simple in-scope consistent-with-other-code suggestion improving related code safety/clarity/maintainability instead of wasting time on ideas that cause those bugs in the first place!